app/components/files/audio_file.tsx

```diff
@@ -0,0 +1,218 @@
+    const style = getStyleSheet(theme);
+    const [hasPaused, setHasPaused] = useState<boolean>(true);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const [hasEnded, setHasEnded] = useState<boolean>(false);
+    const [progress, setProgress] = useState<number>(0);
+    const [timeInMinutes, setTimeInMinutes] = useState<string>('0:00');
+    const [duration, setDuration] = useState<number>(0);
+    const videoRef = useRef<VideoRef>(null);
+
+    useEffect(() => {
+        if (hasEnded) {
+            const timer = setTimeout(() => {
+                setHasPaused(true);
+                setProgress(0);
+                setHasEnded(false);
+            }, 100);
+
+            return () => clearTimeout(timer);
+        }
+
+        return () => null;
+    }, [hasEnded]);
+
+    const source = useMemo(() => ({uri: file.uri}), [file.uri]);
+
+    const {toggleDownloadAndPreview} = useDownloadFileAndPreview();
+
+    const onPlayPress = () => {
+        setHasPaused(!hasPaused);
+    };
+
+    const onDownloadPress = async () => {
+        if (!canDownloadFiles) {
+            alertDownloadDocumentDisabled(intl);
+            return;
+        }
+
+        toggleDownloadAndPreview(file);
+    };
+
+    const loadTimeInMinutes = (timeInSeconds: number) => {
+        const minutes = Math.floor(timeInSeconds / 60);
+        const seconds = Math.floor(timeInSeconds % 60);
+        setTimeInMinutes(`${minutes}:${seconds.toString().padStart(2, '0')}`);
+    };
+
+    const onLoad = (loadData: OnLoadData) => {
+        loadTimeInMinutes(loadData.duration);
+        setDuration(loadData.duration);
+    };
+
+    const onProgress = (progressData: OnProgressData) => {
+        if (hasPaused) {
+            return;
+        }
+        const {currentTime, playableDuration} = progressData;
+        setProgress(currentTime / playableDuration);
+        throttle(() => loadTimeInMinutes(currentTime), 1000)();
+    };
+
+    const onEnd = () => {
+        if (videoRef.current) {
+            videoRef.current.seek(0);
+        }
+        setHasEnded(true);
+    };
+
+    const onError = () => {
+        setHasError(true);
+    };
+
+    const onSeek = (seekPosition: number) => {
+        if (videoRef.current && duration > 0) {
+            const newTime = seekPosition * duration;
+            videoRef.current.seek(newTime);
+            setProgress(seekPosition);
+            throttle(() => loadTimeInMinutes(newTime), 100)();
+        }
+    };
+
+    const onAudioFocusChanged = ({hasAudioFocus}: {hasAudioFocus: boolean}) => {
+        if (!hasAudioFocus) {
+            setHasPaused(true);
+        }
+    };
+
+    if (hasError) {
+        return (
+            <FormattedText
+                id={'audio.loading_error'}
+                defaultMessage={'Error loading audio.'}
+            />
+        );
+    }
+
+    return (
+        <TouchableWithoutFeedback
+            onPress={(e) => {
+                e.stopPropagation();
+            }}
```

Comment:
This creates a new function on every render. Please use `useCallback`.

Reply:
done!